### DISCUSSION TITLE
Statusline for kitty?

### DISCUSSION THREAD

**Discussion by prajwal296** [kovidgoyal/kitty#3226] [Q&A]
Hey, kudos to Kovid and other contributors for developing this amazing project!

I'm trying to build a tmux/vim-like statusline for kitty using `kittens`, to display some information like my current git branch or cpu/ram usage without cluttering my shell prompt. Pardon me if I sound dumb, but I have a couple of questions:
1. Do kittens run on startup? So far the documentation has been great, and explains how I can key-bind a kitten. But no matter what I try, I can't get my script to run on startup.
2. Can kittens be even used to build something like this in first place? Utilizing python's curses module sounds like a great deal to me.

This is just a simple idea and I may end up scrapping it altogether, but any help is welcome.

**Comment by kovidgoyal** [comment]
No kittens cannot be made to run at startup, if you want to do that, use the --session command line argument. launch --allow-remote-control kitty +kitten mykitten.py

 Also note that there are a couple of limitations:

1) kittens cannot get events from kitty, so you would need to poll for status

2) currently there is no way to really have docked windows in kitty, there is an issue open for it, see the issues.

**Reply by MarkJaroski** [reply]
Hi @kovidgoyal , first thank you for your work on both Kitty and Calibre.

My related question is this: there's a status line in your terminal in the intro video. How does that work? Is it powerline? Or something else?

**Comment by kovidgoyal** [comment]
It is https://sw.kovidgoyal.net/kitty/kittens/panel/ with my own custom
code to draw it.

**Reply by MarkJaroski** [reply]
Thank you sir !
